lib/puppet/type/file/content_uri.rb

```diff
@@ -0,0 +1,42 @@
+require 'net/http'
+require 'uri'
+
+module Puppet
+  Puppet::Type.type(:file).newparam(:content_uri) do
+
+    desc <<-'EOT'
+      A unique identifier for file content, which must be a URI which points
+      to a file in a module. This parameter allows the content of a file resource
+      to remain consistent regardless of code updates on the master.
```

Comment:
For Burnside we're only planning on having it serve "files" directories from modules in an environment. We've talked about using the same `code_content` endpoint for other use cases, but there's some debate about whether to reuse the same endpoint or create a new one.